src/bootstrap/builder.rs

```diff
@@ -840,7 +840,13 @@ impl<'a> Builder<'a> {
             // already-passed -C metadata flag with our own. Our rustc.rs
             // wrapper around the actual rustc will detect -C metadata being
             // passed and frob it with this extra string we're passing in.
```

Comment:
So the wording above was correct where the compiler was the only crate graph depending on crates.io, but since now std itself also does that, we need to apply this metadata mangling to differ std's and rustc's symbols as well. Let me know if you want further explanation or you think that's enough to update the wording.